File: culturaConectaREC-ui/src/app/services/api/secretaria/isecretaria.service.ts
import { Observable } from "rxjs";
import { SaveSecretariaDto, SecretariaDto } from "./secretaria.models";

export interface ISecretariaService{

    save(request: SaveSecretariaDto): Observable<SaveSecretariaDto>

    update(id: number, request: SecretariaDto): Observable<SecretariaDto>

    delete(id: number): Observable<void>

    list(): Observable<SecretariaDto[]>

    findByID(id: number): Observable<SecretariaDto>
}

File: culturaConectaREC-api/src/main/java/cesar/ccr/com/service/ISecretariaService.java
package cesar.ccr.com.service;

import cesar.ccr.com.entity.Secretaria;

public interface ISecretariaService{

	Secretaria save(final Secretaria entity);
	
	Secretaria update(final long id, final Secretaria entity);
	
	void delete(final long id);
}
